internal/juju/applications.go

```diff
@@ -637,9 +637,18 @@ func (c applicationsClient) ReadApplication(input *ReadApplicationInput) (*ReadA
 		exposed["cidrs"] = cidrs
 	}
 
+	// (juanmanuel-tirado) In certain scenarios depending on how
+	// charms are uploaded to CharmHub, Juju may register the
+	// latest/stable channel as stable. We check this case here
+	// and force it to be latest/stable
+	normalizedChannel := appInfo.Channel
+	if normalizedChannel == "stable" {
+		normalizedChannel = "latest/stable"
```

Comment:
This seems incorrect. We shouldn't need to coerce the track in the client. The app is deployed with the channel `stable` and `ReadApplication` (`applicationAPIClient.ApplicationsInfo` really) is correctly bringing the channel as `stable`. I think the problem is we [incorrectly default to `latest/stable`](https://github.com/juju/terraform-provider-juju/blob/a4a7e437300b57ae2df2583a864a979d15a2ca53/internal/provider/resource_application.go#LL56C1-L56C13) when a channel is not specified. I believe the correct fix is defaulting to just `stable` instead of fixating the track as well.